apps/emqx/src/emqx_schema.erl

```diff
@@ -2964,24 +2980,74 @@ do_parse_server(Str, Opts) ->
         false ->
             ok
     end,
+    case is_list(DefaultScheme) andalso (not lists:member(DefaultScheme, SupportedSchemes)) of
+        true ->
+            %% inconsistent schema
+            error("bad_schema");
+        false ->
+            ok
+    end,
     %% do not split with space, there should be no space allowed between host and port
     case string:tokens(Str, ":") of
```

Comment:
nit: this is a huge function that perhaps could be split into some small functions, like split at here, and use pattern match to handle the tokens

Reply:
Good point.  I've refactored the function body into a helper one like you suggested, and changed the return type to be a more semantic map.